apps/web/src/app/(signing)/sign/[token]/sign-dialog.tsx

```diff
@@ -13,30 +14,52 @@ export type SignDialogProps = {
   isSubmitting: boolean;
   document: Document;
   fields: Field[];
+  enableTips: () => void;
   onSignatureComplete: () => void | Promise<void>;
 };
 
 export const SignDialog = ({
   isSubmitting,
   document,
   fields,
+  enableTips,
   onSignatureComplete,
 }: SignDialogProps) => {
   const [showDialog, setShowDialog] = useState(false);
+  const [isFieldsSigned, setIsFieldsSigned] = useState(false);
 
   const isComplete = fields.every((field) => field.inserted);
 
+  const fieldSignCheck = () => {
+    const isAllFieldSigned = validateFieldsInserted(fields);
+    if (isAllFieldSigned) {
+      setIsFieldsSigned(true);
+    } else {
+    }
+  };
+
+  useEffect(() => {
+    if (fields.every((field) => field.inserted)) {
+      setIsFieldsSigned(true);
+    } else {
+      setIsFieldsSigned(false);
+    }
+  }, [fields]);
+
   return (
-    <Dialog open={showDialog} onOpenChange={setShowDialog}>
+    <Dialog open={isFieldsSigned && showDialog} onOpenChange={setShowDialog}>
```

Comment:
I think doing `isFieldsSigned && showDialog` will cause some unintended actions where:

1. User has a page open with fields that are not yet inserted
2. User complete
3. User clicks all the fields

When clicking the final field, this dialog will suddenly open

This won't happen with the `onTriggerClick` suggestion above